packages/state-transition/src/cache/epochCache.ts

```diff
@@ -794,15 +899,37 @@ export class EpochCache {
     return shuffling;
   }
 
+  getShufflingDecisionRoot(epoch: Epoch): RootHex {
+    switch (epoch) {
+      case this.epoch - 1:
+        return this.previousDecisionRoot;
+      case this.epoch:
+        return this.currentDecisionRoot;
+      case this.nextEpoch:
+        return this.nextDecisionRoot;
+      default:
+        throw new EpochCacheError({
+          code: EpochCacheErrorCode.DECISION_ROOT_EPOCH_OUT_OF_RANGE,
+          currentEpoch: this.epoch,
+          requestedEpoch: epoch,
+        });
+    }
+  }
+
   getShufflingAtEpochOrNull(epoch: Epoch): EpochShuffling | null {
```

Comment:
could this be safely removed from state-transition, as we use ShufflingCache instead?

Reply:
During tests where the shufflingCache isnt used those would always be `undefined/null` and we would need to duplicate the logic so i centralized into that function.